### PR TITLE
Roselkisma/flaskpg

### PR DESCRIPTION
**Pull request by Roselkisma**
I used useSearchData to get some numbers from the Elasticsearch to create a table about the donor's race. #1930

### CODE REVIEW

File: end-to-end/artillery/queries/datasets.json [hunk 1]
```diff
@@ -65,9 +65,9 @@
       "filter": { "term": { "entity_type.keyword": "Dataset" } },
       "aggs": {
         "donor.mapped_metadata.sex.keyword": {
-          "terms": { "field": "donor.mapped_metadata.sex.keyword", "size": 5 }
+          "terms": { "field": "donor..keyword", "size": 5 }
```
Comment (john-conroy) [inline end-to-end/artillery/queries/datasets.json:68]: I imagine this got changed while you were looking for example queries. These are queries used in load testing that should be kept as is. I would revert these changes.

File: CHANGELOG-table.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- I created a table about the donor's race.
```
Comment (mccalluc) [inline CHANGELOG-table.md:1]:
```suggestion
- Created a table about the donor's race.
```
Style: In the context of the changelog, we usually just have lots of sentence fragments. The "I" doesn't matter.

File: CHANGELOG-hello.md [hunk 3]
```diff
@@ -0,0 +1 @@
+- added Hello World to a page.
```
Comment (mccalluc) [inline CHANGELOG-hello.md:1]: Get rid of this file: only need one CHANGELOG per PR.

File: context/app/static/js/pages/Vis/Vis.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+             "terms": {
+                 "field": "mapped_metadata.sex.keyword"
+             }
+         }
+     }
+ }
+
+ */
+
+const donorSexQuery = {
```
Comment (mccalluc) [inline context/app/static/js/pages/Vis/Vis.jsx:36]: Either race or sex is a fine example to start with, but make sure the variable name matches the content.

File: context/app/static/js/pages/Vis/Vis.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      terms: {
+        field: 'mapped_metadata.race.keyword',
+      },
+    },
+  },
+};
+
+// change to match the data
+const columns = [
+  { id: 'whie', label: 'White' },
```
Comment (mccalluc) [inline context/app/static/js/pages/Vis/Vis.jsx:58]:
```suggestion
  { id: 'white', label: 'White' },
```

File: context/app/static/js/pages/Vis/Vis.jsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  { id: 'black', label: 'Black or African American' },
+];
+
+function Vis() {
+  const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
+  // eslint-disable-next-line no-unused-vars
```
Comment (mccalluc) [inline context/app/static/js/pages/Vis/Vis.jsx:64]:
```suggestion
```
Maybe searchData wasn't used in an earlier version? shouldn't need to disable the warning now.

File: context/app/static/js/pages/Vis/Vis.jsx (hunk 5 above)
Comment (mccalluc) [reply context/app/static/js/pages/Vis/Vis.jsx:58]: ... but see comments below.

File: context/app/static/js/pages/Vis/Vis.jsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+              <TableRow>
+                {columns.map((column) => (
+                  <HeaderCell key={column.id}>{column.label}</HeaderCell>
+                ))}
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              <TableRow>
+                <TableCell> {searchData.aggregations['mapped_metadata.race'].buckets[0].doc_count}</TableCell>
+                <TableCell> {searchData.aggregations['mapped_metadata.race'].buckets[1].doc_count}</TableCell>
```
Comment (mccalluc) [inline context/app/static/js/pages/Vis/Vis.jsx:82]: You assume that the order of the columns you've given above will match the order of the buckets... with more data, it might not. You need to pull the label for the bucket from the `searchData`, rather than hard coding.

And then in `HeaderCell`, it would be fine to use the `label` for the `key`... The `key` just needs to be unique, and when you have pulled the bucket names from `searchData`, they will be.

File: end-to-end/artillery/queries/datasets.json [hunk 8]
```diff
@@ -65,9 +65,9 @@
...
         },
-        "donor.mapped_metadata.sex.keyword_count": {
+        "donor..keyword_count": {
```
Comment (mccalluc) [inline end-to-end/artillery/queries/datasets.json:70]: I don't think you want to touch this file at all. Revert to original.

**Review by mccalluc — CHANGES_REQUESTED** [review]
Lots of little changes, but looks like the right direction. CI is complaining about python formatting:

```
  ./context/app/routes_main.py:216:1: E302 expected 2 blank lines, found 1
  ./context/app/routes_main.py:230:1: E302 expected 2 blank lines, found 1
  Try: autopep8 --in-place --aggressive -r . --exclude node_modules,ingest-validation-tools
  Error
```

File: context/app/routes_main.py [hunk 9]
```diff
@@ -209,6 +209,19 @@ def dev_search():
         flask_data=flask_data
     )
```
Comment (mccalluc) [inline context/app/routes_main.py:211]:
```suggestion


```

File: context/app/routes_main.py [hunk 10]
```diff
@@ -209,6 +209,19 @@ def dev_search():
...
+    flask_data = {
+        'endpoints': _get_endpoints(),
+        'title': title
+    }
+    return render_template(
+        'pages/base_react.html',
+        title=title,
+        types=entity_types,
+        flask_data=flask_data
+    )
```
Comment (mccalluc) [inline context/app/routes_main.py:225]:
```suggestion


```

**Review by mccalluc — CHANGES_REQUESTED** [review]
This is fix the pyflake problem.

File: end-to-end/artillery/queries/datasets.json (hunk 8 above)
Comment (Roselkisma) [reply end-to-end/artillery/queries/datasets.json:70]: I did not touch this file. This is how it looks on my computer. I do not know why it looks different after I push the changes to github, because I did not do anything to this file. 
 
Screen Shot 2021-07-07 at 1 06 19 PM

File: end-to-end/artillery/queries/datasets.json (hunk 1 above)
Comment (Roselkisma) [reply end-to-end/artillery/queries/datasets.json:68]: I did not touch this file. This is how it looks on my computer. I do not know why it looks different after I push the changes to github, because I did not do anything to this file. 
 
Screen Shot 2021-07-07 at 1 06 19 PM

File: end-to-end/artillery/queries/datasets.json (hunk 8 above)
Comment (john-conroy) [reply end-to-end/artillery/queries/datasets.json:70]: If that's the case, you can use `git pull` to pull down the current code and change it back and commit it afterwards.

Comment (john-conroy) [reply end-to-end/artillery/queries/datasets.json:70]: There are other ways to do it, but that's probably the most straightforward.

**Review by Roselkisma — COMMENTED** [review]
I made all the changes

File: end-to-end/artillery/queries/datasets.json (hunk 1 above)
Comment (mccalluc) [reply end-to-end/artillery/queries/datasets.json:68]: @Roselkisma -- We understand that this is what your local version has, the problem is that this is a change from what is in master. (Red is master / green is yours.) It was an accident, it just needs to be undone. Not a big deal.

**Review by mccalluc — CHANGES_REQUESTED** [review]
I think there is some confusion. Let's find time to talk?

File: end-to-end/artillery/queries/datasets.json (hunk 1 above)
Comment (mccalluc) [inline end-to-end/artillery/queries/datasets.json:68]:
```suggestion
          "terms": { "field": "donor.mapped_metadata.sex.keyword", "size": 5 }
```

File: end-to-end/artillery/queries/datasets.json (hunk 8 above)
Comment (mccalluc) [inline end-to-end/artillery/queries/datasets.json:70]:
```suggestion
        "donor.mapped_metadata.sex.keyword_count": {
```

**Review by mccalluc — CHANGES_REQUESTED** [review]
@Roselkisma -- Here are the last two changes that we discussed. If it's ok, I'd like to get this PR merged so we switch the default branch.

**Review by mccalluc — APPROVED** [review]
